### PR TITLE
Use the process environment when spawning apm. 

### PR DESCRIPTION
**Pull request by simtel12**
Also allow the setting of the app directory for rebuilding modules - this allows an arbitrary location for the app in relation to this task.

### CODE REVIEW

File: tasks/download-atom-shell-task.coffee [hunk 1]
```diff
@@ -93,24 +93,33 @@ module.exports = (grunt) ->
...
-      env = ATOM_NODE_VERSION: currentVersion.substr(1)
-      spawn {cmd: apm, args: ['rebuild'], env}, callback
+
+      # When we spawn apm, we still want to use the global environment variables
+      options = env: process.env
+      options.env.ATOM_NODE_VERSION = currentVersion.substr(1)
+
+      # If the appDir has been set, then that is where we want to perform the rebuild.
+      # it defaults to the current directory
+      if (appDir?)
```
Comment (kevinsawicki) [inline tasks/download-atom-shell-task.coffee]: these parens aren't needed

Comment (simtel12) [inline tasks/download-atom-shell-task.coffee]: Ah, thanks for the note.  I'll get that fixed up.

**Comment by kevinsawicki** [comment]
Thanks for this

**Comment by kevinsawicki** [comment]
Published in 2.1.0

**Pull request merged**